stack/update_all_stacks.py

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env python
+
+"""Updates all stacks in an order compatible with each stack dependency."""
+
+import glob
+import graphlib  # TopologicalSorter requires python >= 3.9.
+import os
+import subprocess
+import yaml
+
+deps = {}
+for filename in glob.glob('Pulumi.*.yaml'):
+    with open(filename) as f:
+        parsed = yaml.safe_load(f)
+    dataset = filename.split('.')[1]
+    deps[dataset] = parsed['config'].get('datasets:depends_on', [])
+    if deps[dataset]:
+        # Parse the string representation of the list.
+        deps[dataset] = yaml.safe_load(deps[dataset])
+
+env = dict(os.environ, PULUMI_CONFIG_PASSPHRASE='')
+for dataset in graphlib.TopologicalSorter(deps).static_order():
```

Comment:
Oh that's cool, didn't know that existed, though how many people have python >= 3.9...